packages/design-system/src/components/web-components/ds-drawer/ds-drawer.stories.tsx

```diff
@@ -0,0 +1,195 @@
+      non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor
+      sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
+      magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+      aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit
+      esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident,
+      sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet,
+      consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna
+      aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex
+      ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
+      dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa
+      qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur
+      adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
+      minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
+      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu
+      fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
+      officia deserunt mollit anim id est laborum.
+    </p>
+  </>
+);
+
+const Template = (args) => {
+  const [drawerOpen, setDrawerOpen] = useState(args['is-open'] ?? false);
+
+  useEffect(() => {
+    if (args['is-open'] !== drawerOpen) {
+      setDrawerOpen(args['is-open'] ?? false);
+    }
+  }, []);
+
+  useEffect(() => {
+    const drawerElement = document.querySelector('ds-drawer');
+    const toggleButton = document.querySelector('.ds-c-drawer__toggle');
+
+    if (drawerElement && toggleButton) {
+      const handleDrawerClose = (event: Event) => {
+        action('ds-close-click')(event);
+        setDrawerOpen(false);
+      };
+
+      const handleDrawerOpen = () => {
+        setDrawerOpen(true);
+      };
+
+      drawerElement.addEventListener('ds-close-click', handleDrawerClose as EventListener);
+      toggleButton.addEventListener('click', handleDrawerOpen);
+
+      return () => {
+        drawerElement.removeEventListener('ds-close-click', handleDrawerClose as EventListener);
+        toggleButton.removeEventListener('click', handleDrawerOpen);
+      };
+    }
+  }, []);
+
+  const formattedArgs = {
+    ...args,
+    'is-open': drawerOpen.toString(),
+  };
+
+  const toggleButtonArgs = {
+    'class-name': 'ds-c-drawer__toggle',
+    variation: 'ghost',
+  };
+
+  return (
+    <div>
+      <ds-drawer {...formattedArgs}>
+        {drawerContent}
+        {args['footer-body'] && (
```

Comment:
@jack-ryan-nava-pbc, I agree that preactment is handling the slots and corresponding attributes in a mutually exclusive way, which results in the slot content being prioritized when both are provided.

For this storybook configuration, I followed a pattern I saw in the story file for [ds-choice](https://github.com/CMSgov/design-system/blob/main/packages/design-system/src/components/web-components/ds-choice/ds-choice.stories.tsx#L171-L175) I thought it might be good to demonstrate both options—using the `footer-body` attribute and providing content through the slot—so that storybook users could interact with the control directly.
But maybe this is too misleading?

Reply:
Huh, I'm not sure. It doesn't look like the way the storybook is set up for ds-choice is completely accurate to the way the component logic works. I think `checked` and `unchecked` children are mutually exclusive in the React component? But I might be wrong. 

I think it's a little confusing in that Storybook makes it look like the two things are dependent on each other when in fact they aren't. Maybe @pwolfert can offer some context into the way the `ds-choice` story is wired up?